lib/galaxy/webapps/galaxy/api/jobs.py

```diff
@@ -97,6 +98,17 @@ def index(self, trans: ProvidesUserContext, **kwd):
         :type   user_details: boolean
         :param  user_details: if true, and requestor is an admin, will return external job id and user email.
 
+        :type   user_id: str
+        :param  user_id: an encoded user id to restrict query to, must be own id if not admin user
+
+        :type   limit: int
+        :param  limit: Maximum number of jobs to return.
```

Comment:
We should probably provide a upper limit here that can be returned in a reasonable short amount of time. Maybe 1000 is a good value ?

Reply:
It's 500 for `/api/datasets`, maybe stick with that for consistency?